numpy/core/src/multiarray/alloc.c

```diff
@@ -113,8 +113,11 @@ npy_free_cache(void * p, npy_uintp sz)
 NPY_NO_EXPORT void *
 npy_alloc_cache_dim(npy_uintp sz)
 {
-    /* dims + strides */
-    if (NPY_UNLIKELY(sz < 2)) {
+    /*
+     * make sure any temporary allocation can be used for array metadata which
+     * uses one memory block for both dimensions and strides
+     */
```

Comment:
On a similar note, we'll be missing the existing cache a lot more here since dims alone can be odd - is it worth looking always looking up in the cache with `sz | 1` or something, so that the lookup is always odd?